### ISSUE TITLE
remote BAM

### ISSUE THREAD

**Issue by daler** [arq5x/bedtools2#258]
Hi Aaron & Neil - 

I thought that remote BAM support worked at some point but I'm unable to get it working now.

samtools finds the remote BAM:
```bash
URL="http://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam"
samtools view $URL | head -n2
#[knet_seek] SEEK_END is not supported for HTTP. Offset is unchanged.
#[bam_header_read] EOF marker is absent. The input is probably truncated.
#SRR010939.15011799	35	21	33019936	99	76M	=	33019947	87	TAAAGATATAATCAGTAACTAAACTAATCCCAACACTAGGATTATTTGCCTAAATCATATATATGTGTATAGAGAA	=789=7D69;<E<5?A%7.D=;1AC1,ED,CA;<:0E>?>&BB6<H@?*7A/99C82<141C6C8575-.-$+$*,RG:Z:SRR010939	MF:i:18	Aq:i:77	NM:i:2	UQ:i:15	H0:i:1	H1:i:0
#SRR010939.15011799	19	21	33019947	99	76M	=	33019936	-87	TCAGTAACTCAGCTAATCCCAACACTAGGATTATTTGCCTAAATCATATATATGTGTATATATAATCATATACTTT	,/06.-856$E*?*D-*=??EC@C=1D@8A=:D;6>,9?>FG+7@C9E5C>DA?3-;F6F0D<DD/?D:B1A=623RG:Z:SRR010939	MF:i:18	Aq:i:77	NM:i:2	UQ:i:19	H0:i:1	H1:i:0
```

But bedtools just hangs. It's using CPU, but there's no output:
```bash
bedtools --version
#bedtools v2.24.0-12-ge1993b5

# hangs
bedtools bamtobed -i $URL | head -n2
```

While it's hanging, if I do this:
```
lsof | grep bedtools | TCP
```
then the last field says:
```
[...my hostname...]->hgw3.cse.ucsc.edu:http (CLOSE_WAIT)
```

Any ideas? Thanks.

**Comment by nkindlon** [comment]
Hi Ryan,
    Aaron is largely indsiposed this week, but I took a look at this. Remote files are definitely not something I ever added or accounted for in the tools I worked on. I was just reading a bit online about how that can be handled, but it's not something we ever did as far as I know. It's helpful that you picked bamtobed for the example. From the history, that was last touched by Aaron in Dec 2013, and shows no changes involving remote files that I see.

I did find this thread discussion between Aaron and Derek Barnett, the developer of BamTools (which we just call in our code), regarding adding remote file support:
https://github.com/pezmaster31/bamtools/issues/11
Not sure what came out of that since; perhaps there's now something we can add as an enhancement.

**Comment by daler** [comment]
Hi Neil - 

Thanks for checking into this. I ran across the issue as part of an overhaul of pybedtools -- I couldn't get the previously-passing remote bam test to work.

Now after looking at it more closely, the previous pybedtools support for remote bams was really just slurping in remote files via samtools and then saving them locally to a tempfile. So it was somewhat of a hack, and not really using the remote functionality. I had removed that hack in the new version, passing the remote file directly to bedtools -- but causing the test to fail.

Remote file support would be nice to have eventually. Luckily, for now I can just say "bedtools doesn't support remote bams, so neither does pybedtools"!
